src/NaCl.Core/Base/ChaCha20BaseIntrinsics.cs

```diff
@@ -0,0 +1,69 @@
+﻿#if INTRINSICS
```

Comment:
I wonder if we do need the preprocessor directive since we are checking if the hardware supports it